src/gui/qgscolorbutton.cpp

```diff
@@ -443,14 +495,21 @@ QPixmap QgsColorButton::createMenuIcon( const QColor &color, const bool showChec
 
 void QgsColorButton::buttonClicked()
 {
-  switch ( mBehavior )
+  if ( !mLinkedColorName.isEmpty() )
```

Comment:
For the sake of consistency, we should probably settled on either mLinkedColorName.isEmpty() or  linkedProjectColor().isValid() as a conditional check for a linked project color. Whichever is less costly :) see e.g. QgsColorButton::dragEnterEvent for the alternate check.